src/main/com/aca/homework/week7/clickable/AbstractClickable.java

```diff
@@ -0,0 +1,21 @@
+package com.aca.homework.week7.clickable;
+
+public abstract class AbstractClickable implements Clickable {
+
+    private Action action;
+
+    public AbstractClickable() {
```

Comment:
because the name is common for all subtypes, please implement name logic in this class